test/fake-api-db/index.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+const rulesJson = require('../fake-api-fixtures/rules/index.json');
+const rulesFilePath = path.join(__dirname, 'db-rules.json');
+
+const seed = (filePath, data) => fs.outputJson(filePath, data);
+const resetState = () => {
+  return Promise.all([
+    seed(rulesFilePath, rulesJson)
+    // add seeding to reset other state
+  ]);
+};
+
+class FakeDb {
+  constructor (filePath) {
+    this.filePath = filePath;
+  }
+
+  addItem (record) {
+    return fs.readJson(this.filePath)
+      .then((data) => {
+        data.meta.count += 1;
+        data.results.push(record);
+        return data;
+      })
+      .then((data) => seed(this.filePath, data));
+  }
+
+  getItems () {
+    return fs.readJson(this.filePath);
+  }
+
+  deleteItem (name) {
```

Comment:
This only works for `rule`

Reply:
Yeah, I thought that storage for other types of data could override. Would you prefer I move this to the `FakeRulesDb` class?

Reply:
yes.  delete collection for example has `name and version` as argument.

Reply:
OK. I moved it to the `FakeRulesDb` class